core/arch/arm/sm/sm_a32.S

```diff
@@ -268,6 +268,33 @@ UNWIND(	.fnstart)
 	cps	#CPSR_MODE_MON
 	/* Point just beyond sm_ctx.sec */
 	sub	sp, r0, #(SM_CTX_SIZE - SM_CTX_NSEC)
+
+#ifdef CFG_INIT_CNTVOFF
+	read_scr r0
+	orr	r0, r0, #SCR_NS /* Set NS bit in SCR */
+	write_scr r0
+
+	/*
+	 * Accessing CNTVOFF:
+	 * If the implementation includes the Virtualization Extensions
+	 * this is a RW register, accessible from Hyp mode, and
+	 * from Monitor mode when SCR.NS is set to 1.
+	 * If the implementation includes the Security Extensions
+	 * but not the Virtualization Extensions, an MCRR or MRRC to
+	 * the CNTVOFF encoding is UNPREDICTABLE if executed in Monitor
+	 * mode, regardless of the value of SCR.NS.
+	 * So test VIRT and GENTIMER Mask both here.
+	 */
+	read_idpfr1 r2
+	ands	r2, r2, #(IDPFR1_GENTIMER_MASK | IDPFR1_VIRT_MASK)
+	movne	r2, #0
+	/* Reset CNTVOFF to 0 */
+	mcrrne	p15, 4, r2, r2, c14
```

Comment:
Can be define a macro `write_cntvoff` from arm32_macros.S ? (comment a previous line would become redundant).

Reply:
mcrr could be used in write_cntvoff, but I am afraid mcrrne could not.

Reply:
On the other hand, this isn't optimized code so we could perhaps bear the cost of a `bne` for the benefit of more readable code.